client/consensus/common/src/parachain_consensus.rs

```diff
@@ -346,6 +346,18 @@ where
 	P: UsageProvider<Block> + Send + Sync + BlockBackend<Block>,
 	for<'a> &'a P: BlockImport<Block>,
 {
+	let best_number = parachain.usage_info().chain.best_number;
+	if *header.number() < best_number {
```

Comment:
@bkchr Could there be the case where a re-org on the relay would actually create a re-org on the parachain leading to a lower block number becoming best ?
I'm thinking in the case where a re-org would happen over 2 blocks on the relay, and the parachain network would have only produced 1 block in the fork branch.

Reply:
Then it will take one block more to set a new best block and you see the reorg later.

Reply:
I know this is probably a rare case, but what if the node is the selected author for the new block after this re-org.
The node would not import the new fork block and so will not produce a new one ?

(This can/should be solved by other mechanism like selecting authors based on other input that just the previous block, but it still remains a possible issue)

Reply:
This is certainly not a perfect solution and I think you are correct that it may lead to missed authorship opportunities during reorgs.

Reply:
We will probably need changes in Substrate client to avoid this entire issue, but it is without a doubt possible to improve this and future client versions will address it.

Reply:
which block would not be imported?

This here has nothing to do with authorship. The best block is also not important for block authorship

Reply:
This change here is entirely about which block will be set as new best block. And the bug we had seen will only happen on major sync, but never when we are syncing at the tip of the chain.